lib/spack/docs/getting_started.rst

```diff
@@ -38,14 +61,46 @@ For a richer experience, use Spack's `shell support
 
 This automatically adds Spack to your ``PATH``.
 
-------------
-Installation
-------------
+^^^^^^^^^^^^^^^^^
+Clean Environment
+^^^^^^^^^^^^^^^^^
+
+Many packages' installs can be broken by changing environment
+variables.  For example, a package might pick up the wrong build-time
+dependencies (most of them not specified) depending on the setting of
+``PATH``.  ``GCC`` seems to be particularly vulnerable to these issues.
 
-You don't need to install Spack; it's ready to run as soon as you
-clone it from git.
+Therefore, it is recommended that Spack users run with a *clean
+environment*, especially for ``PATH``.  Only software that comes with
+the system, or that you know you wish to use with Spack, should be
+included.  This procedure will avoid many strange build errors.
 
-You may want to run it out of a prefix other than the git repository
+
+^^^^^^^^^^^^^^^^^^
+Check Installation
+^^^^^^^^^^^^^^^^^^
+
+With Spack installed, you should be able to run some basic Spack
+commands.  For example:
+
+.. code-block:: console
+
+    $ spack spec netcdf
+      ...
+      netcdf@4.4.1%gcc@5.3.0~hdf4+mpi arch=linux-SuSE11-x86_64
+          ^curl@7.50.1%gcc@5.3.0 arch=linux-SuSE11-x86_64
+              ^openssl@system%gcc@5.3.0 arch=linux-SuSE11-x86_64
+              ^zlib@1.2.8%gcc@5.3.0 arch=linux-SuSE11-x86_64
+          ^hdf5@1.10.0-patch1%gcc@5.3.0+cxx~debug+fortran+mpi+shared~szip~threadsafe arch=linux-SuSE11-x86_64
+              ^openmpi@1.10.1%gcc@5.3.0~mxm~pmi~psm~psm2~slurm~sqlite3~thread_multiple~tm+verbs+vt arch=linux-SuSE11-x86_64
+          ^m4@1.4.17%gcc@5.3.0+sigsegv arch=linux-SuSE11-x86_64
+              ^libsigsegv@2.10%gcc@5.3.0 arch=linux-SuSE11-x86_64
```

Comment:
That's a cool command, but I'll keep it the way it is for now.  The `...`
in the output is stuff I manually removed.  Often, computer programs or
output need to be "shortened" to work well in a document.

Pedagogically, it doesn't matter if versions change in the future.

On Wed, Sep 7, 2016 at 3:39 PM, Adam J. Stewart <notifications@github.com>
wrote:

> In lib/spack/docs/getting_started.rst:
>
> > +
> > +With Spack installed, you should be able to run some basic Spack
> > +commands.  For example:
> > +
> > +.. code-block:: console
> > +
> > +    $ spack spec netcdf
> > +      ...
> > +      netcdf@4.4.1%gcc@5.3.0~hdf4+mpi arch=linux-SuSE11-x86_64
> > +          ^curl@7.50.1%gcc@5.3.0 arch=linux-SuSE11-x86_64
> > +              ^openssl@system%gcc@5.3.0 arch=linux-SuSE11-x86_64
> > +              ^zlib@1.2.8%gcc@5.3.0 arch=linux-SuSE11-x86_64
> > +          ^hdf5@1.10.0-patch1%gcc@5.3.0+cxx~debug+fortran+mpi+shared~szip~threadsafe arch=linux-SuSE11-x86_64
> > +              ^openmpi@1.10.1%gcc@5.3.0~mxm~pmi~psm~psm2~slurm~sqlite3~thread_multiple~tm+verbs+vt arch=linux-SuSE11-x86_64
> > +          ^m4@1.4.17%gcc@5.3.0+sigsegv arch=linux-SuSE11-x86_64
> > +              ^libsigsegv@2.10%gcc@5.3.0 arch=linux-SuSE11-x86_64
>
> You can replace this entire code block with:
>
> .. command-output:: spack spec netcdf
>
> It automatically runs the command and writes the output to the
> documentation. That way it will update when new version of netcdf come out
> or dependencies get added.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/LLNL/spack/pull/1676/files/c9e83c62a002c42852fcb54a3b45d54a68cb8ccc#r77889575>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB1cd9-nWKk6y4-cgSTHKmPpxQ6fnhyCks5qnxLtgaJpZM4JxOaa>
> .
>